src/handlers/proposalsHandlers.ts

```diff
@@ -56,7 +62,175 @@ const getProposals = (
     });
 };
 
-const getProposal = (
+/**
+ * Merges the given field values into a new copy of the given proposal.
+ * It is OK if there are no versions or zero field values.
+ * @param proposal The proposal with versions but no values.
+ * @param values The values to merge into the given proposal.
+ * @returns A new proposal with the values merged into their respective versions.
+ */
+export const mergeProposalFieldValues = (
+  proposal: Proposal,
+  values: ProposalFieldValue[],
+): Proposal => {
+  const newProposal = structuredClone(proposal);
+  const { versions } = newProposal;
+  if (versions === undefined) {
```

Comment:
I actually think that avoiding the destructuring above might make this check more legible since the context of what is being checked would be right there in the line (though see below for another idea!)

```
if (proposal.versions === undefined)
```

Reply:
I had a feeling that `tsc` is more able to grasp a top-level variable undefinedness check rather than a property on a variable or index into a variable, but I haven't experimented enough to say that with any confidence. I will try it and see.

Reply:
I tried without destructuring and `tsc` was unhappy with it later on (these line numbers are after removing the docstring so do not correspond exactly to what was pushed):
```
src/handlers/proposalsHandlers.ts:79:27 - error TS18048: 'newProposal.versions' is possibly 'undefined'.

79     let proposalVersion = newProposal.versions[versionIndex];
                             ~~~~~~~~~~~~~~~~~~~~

src/handlers/proposalsHandlers.ts:86:27 - error TS18048: 'newProposal.versions' is possibly 'undefined'.

86       if (versionIndex >= newProposal.versions.length) {
                             ~~~~~~~~~~~~~~~~~~~~

src/handlers/proposalsHandlers.ts:89:25 - error TS18048: 'newProposal.versions' is possibly 'undefined'.

89       proposalVersion = newProposal.versions[versionIndex];
                           ~~~~~~~~~~~~~~~~~~~~
```
If I keep the destructuring present but then check `if (newProposal.versions === undefined)` then I get this later on:
```
src/handlers/proposalsHandlers.ts:80:27 - error TS18048: 'versions' is possibly 'undefined'.

80     let proposalVersion = versions[versionIndex];
                             ~~~~~~~~

src/handlers/proposalsHandlers.ts:87:27 - error TS18048: 'versions' is possibly 'undefined'.

87       if (versionIndex >= versions.length) {
                             ~~~~~~~~

src/handlers/proposalsHandlers.ts:90:25 - error TS18048: 'versions' is possibly 'undefined'.

90       proposalVersion = versions[versionIndex];
                           ~~~~~~~~
```
I'd rather not check it four times, nor check it once before destructuring and once after destructuring, so I think what we have there is still least worst.

Reply:
It is still unclear to me which situations `tsc` can deduce successfully potential `undefined` vars and which it cannot. I am still surprised at some of the situations it cannot. And for the purposes of this PR, I think this also is moot with the refactor in c602b6a.